### PR TITLE
Add gallery example "Quoted lines" (`style="q"`)

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to add a gallery example showing how to generated "Quoted lines".
 
**Upstream GMT documentation**: https://docs.generic-mapping-tools.org/dev/plot.html#s

**Related to**: 
- Gallery example [Line fronts](https://www.pygmt.org/dev/gallery/lines/linefronts.html)  (`style="f"`)
- PR #2564

**Preview**:  https://pygmt-dev--2563.org.readthedocs.build/en/2563/gallery/lines/quoted_lines.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/gallery/lines/quoted_lines.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+x = np.arange(0, 10 * np.pi, 0.1)
+y = np.sin(0.8 * x)
+
+fig = pygmt.Figure()
+
+fig.basemap(
+    region=[0, 30, -4, 4],
+    projection="X10c/5c",
+    frame=True,
+)
```
Comment (seisman) [inline examples/gallery/lines/quoted_lines.py:89]: Writting in a single line to reduce the number of lines.
```suggestion
fig.basemap(region=[0, 30, -4, 4], projection="X10c/5c", frame=True)
```

File: examples/gallery/lines/quoted_lines.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+fig.plot(
+    x=x,
+    y=y + 2,
+    style="qd1.2c:+lstraight text+f5p",
+    pen="1p,blue",
+)
```
Comment (seisman) [inline examples/gallery/lines/quoted_lines.py:96]:
```suggestion
fig.plot(x=x, y=y + 2, style="qd1.2c:+lstraight text+f5p", pen="1p,blue")
```

File: examples/gallery/lines/quoted_lines.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+# Generate a two-point line for plotting
+x = np.array([1, 4])
+y = np.array([20, 20])
+
+fig = pygmt.Figure()
+fig.basemap(
+    region=[0, 10, 0, 20],
+    projection="X15c/15c",
+    frame="+tQuoted Lines",
+)
```
Comment (seisman) [inline examples/gallery/lines/quoted_lines.py:26]: Can this be written in a single line?
```suggestion
fig.basemap(region=[0, 10, 0, 20], projection="X15c/15c", frame="+tQuoted Lines")
```

**Review by seisman — APPROVED** [review]

File: examples/gallery/lines/quoted_lines.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
+"""
+Quoted lines
+------------
+To plot a so-called *quoted line*, i.e., labels along a line
+or curve, use the ``style`` parameter of the
+:meth:`pygmt.Figure.plot` method with the argument ``"q"`` and the
+desired modifiers. A colon (``":"``) is used to separate the
+algorithm settings from the label information.
+This example shows how to adjust the labels.
+For modifying the base line via the ``pen`` parameter, see the
```
Comment (michaelgrund) [inline examples/gallery/lines/quoted_lines.py:10]:
```suggestion
For modifying the baseline via the ``pen`` parameter, see the
```

File: examples/gallery/lines/quoted_lines.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        justify="ML",
+        offset="0.75c/0c",
+    )
+
+fig.show()
+
+
+###############################################################################
+# For curved labels following the line, append ``"+v"`` to the argument passed
+# to the `style` parameter.
```
Comment (michaelgrund) [inline examples/gallery/lines/quoted_lines.py:75]:
```suggestion
# to the ``style`` parameter.
```

**Review by michaelgrund — APPROVED** [review]

File: examples/gallery/lines/quoted_lines.py (hunk 4 above)
Comment (yvonnefroehlich) [reply examples/gallery/lines/quoted_lines.py]: All occurencies are updated in commit https://github.com/GenericMappingTools/pygmt/pull/2563/commits/c3b04926e0eaf01d01edee45bde5bda56711cb17.

**Pull request merged**